racket/collects/racket/hash.rkt

```diff
@@ -63,6 +63,12 @@
                     (combine/key k v (hash-ref hm k))))
         res)))
 
+(define (hash-filter-values ht pred)
+  (for/hash ([(k v) (in-hash ht)] #:when (pred v)) (values k v)))
```

Comment:
This will always construct an immutable hash with `equal?` as the key comparator, which doesn't seem right to me.

Reply:
Thanks, I was aware of immutable/mutable but have never delved into different hash comparators. I've updated the function to try to maintain the right configuration and added new tests/examples